pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -744,17 +746,21 @@ func buildGatewayNetworkFiltersFromTCPRoutes(node *model.Proxy, push *model.Push
 			continue
 		}
 
+		if server.GetTls().GetMode() == networking.ServerTLSSettings_ISTIO_MUTUAL {
+			out = append(out, xdsfilters.TCPListenerMx)
```

Comment:
to be able to add MX filter at gateways only in ISTIO_MUTUAL case, we will have to pass tls mode in all the function calls i.e to `buildOutboundNetworkFilters` , `buildOutboundNetworkFiltersWithSingleDestination`, `buildOutboundNetworkFiltersWithSingleDestination` and `buildMetadataExchangeNetworkFilters`
And I guess that would require fixing unit tests as well for this additional argument. To avoid these changes, thought of doing it here itself.

Reply:
My concern is this may cause diverge in the future. 
We should also respect the PILOT_ENABLE_METADATA_EXCHANGE env

How about 

```
if server.GetTls().GetMode() == networking.ServerTLSSettings_ISTIO_MUTUAL {
            buildMetadataExchangeNetworkFilters )() // 
}
```